subsys/nfc/ndef/ch_rec_parser.c

```diff
@@ -0,0 +1,585 @@
+/*
+ * Copyright (c) 2020 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-BSD-5-Clause-Nordic
+ */
+
+#include <errno.h>
+#include <logging/log.h>
+#include <net/buf.h>
+#include <sys/byteorder.h>
+
+#include <nfc/ndef/msg_parser.h>
+#include <nfc/ndef/ch_rec_parser.h>
+#include <nfc/ndef/payload_type_common.h>
+
+#define TAG_TYPE_4_NLEN_FIELD_SIZE 2
+/* Size of the field with CPS data. */
+#define AC_REC_CPS_BYTE_SIZE 1
+
+LOG_MODULE_REGISTER(nfc_ndef_ch_rec_parser,
+		    CONFIG_NFC_NDEF_CH_REC_PARSER_LOG_LEVEL);
+
+static u8_t *memory_allocate(struct net_buf_simple *buf,
+			     u32_t alloc_size)
+{
+	if (net_buf_simple_tailroom(buf) < alloc_size) {
+		return NULL;
+	}
+
+	return net_buf_simple_add(buf, alloc_size);
+}
+
+static bool rec_type_check(const struct nfc_ndef_record_desc *rec_desc,
+			   const u8_t *type, size_t type_len)
+{
+	if (!rec_desc || !rec_desc->tnf) {
+		return false;
+	}
+
+	if (rec_desc->tnf != TNF_WELL_KNOWN) {
+		return false;
+	}
+
+	/* All Connection Handover Record types have the same length. */
+	if (rec_desc->type_length != type_len) {
+		return false;
+	}
+
+	if (memcmp(rec_desc->type, type,
+		   type_len) != 0) {
+		return false;
+	}
+
+	return true;
+}
+
+static int hc_rec_payload_parse(struct nfc_ndef_bin_payload_desc *payload_desc,
+				u8_t *result_buf, u32_t *result_buf_len)
```

Comment:
I think you should update ```result_buf_len``` before function exits

Reply:
Yes, added also in other parsers